### DISCUSSION TITLE
Automatically recognize the directory of odoo

### DISCUSSION THREAD

**Discussion by yao-big** [odoo/odoo-ls#79] [Ideas]
My current situation is like this

I am using the enterprise version of odoo16

The directory of the project is as follows

![image](https://github.com/odoo/odoo-ls/assets/30170943/0638993b-ef86-4a92-9b04-67ac6c5fc8fd)
Configure the odoo plugin in this way
![image](https://github.com/odoo/odoo-ls/assets/30170943/c3287e51-7214-4b04-9737-7e8afaee1d03)
He cannot detect addons, so I need to modify the structure inside the source code of the downloaded ODOO Enterprise Edition, which is not very friendly,

**Comment by yao-big** [comment]
![image](https://github.com/odoo/odoo-ls/assets/30170943/a764f695-08f2-4388-8164-133d501999f8)

This is the expanded directory

**Comment by fda-odoo** [comment]
Hello,

Indeed actually the extension is searching for 'base' module at 'your_odoo_path/odoo/addons' and your addons at 'your_odoo_path/addons'. We will improve that as it only matches the respository structure of Odoo (https://github.com/odoo/odoo/tree/17.0/odoo). 
It is not really explicit too, but you shouldn't have to put your third addon path in the additional addon paths. 'your_addon_path/odoo/addons' is implicit with your odoo path. We can improve the UI about that too

**Reply by yao-big** [reply]
Thank you. Could you please develop this plugin as soon as possible? I think it's still very useful

**Reply by fda-odoo** [reply]
We are working on it ;)

**Reply by jwaes** [reply]
Does not fit the structure of the tar.gz nightly either ...

**Comment by yao-big** [comment]
<img width="1050" alt="image" src="https://github.com/odoo/odoo-ls/assets/30170943/c6da5aa0-9876-4260-9f8b-352dde1e2db7">
There is a bug that keeps generating large log files

**Reply by fda-odoo** [reply]
Please create an issue per subject ;) I moved this comment here: https://github.com/odoo/odoo-ls/issues/82
